app/src/org/commcare/adapters/EntityDetailPagerAdapter.java

```diff
@@ -59,6 +59,11 @@ public Fragment createFragment(int position) {
         return fragment;
     }
 
+    public CharSequence getPageTitle(int position) {
```

Comment:
is there a reason this method is here and not in `TabbedDetailView` itself ?

Reply:
It might be a stretch but the reason is that the adapter is responsible for providing the content of the tabs and can also provide their titles. The structural aspect is that the [displayableChildDetails](https://github.com/dimagi/commcare-android/blob/d3f29f1cc1bce0fdaf14d8c5d4ce00f64c464c6d/app/src/org/commcare/adapters/EntityDetailPagerAdapter.java#L30) which stores the visible detail children and we would need to refactor this logic which I'm not sure if it's worth the effort. Any thoughts?